### PR TITLE
Added the latest debian version (bullseye), now entering 'Hard Freeze' status

### PR DESCRIPTION
**Pull request by lightswitch05**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [ ] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**
*A detailed description, screenshots (if necessary), as well as links to any relevant GitHub issues*

Adding new bullseye Debian version which is entering 'Soft Freeze' status.

**How does this PR accomplish the above?:**
*A detailed description (such as a changelog) and screenshots (if necessary) of the implemented fix*

Added a new target so that existing targets are not changed. Set buster as the default. I don't believe this default tag is even being used anywhere?

**What documentation changes (if any) are needed to support this PR?:**
*A detailed list of any necessary changes*

None.

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.

### CODE REVIEW

**Comment by lightswitch05** [comment]
Failing due to:

```
dpkg: regarding libpam-modules_1.4.0-4_amd64.deb containing libpam-modules:amd64, pre-dependency problem:
 libpam-modules:amd64 pre-depends on libpam-modules-bin (= 1.4.0-4)
  libpam-modules-bin is installed, but is version 1.4.0-2.
```

This error happen on this step: https://github.com/pi-hole/docker-base-images/blob/598df7c7e0b3b64957e467374af2f2bb238a3c31/debian-base/setup.sh#L6-L9

I don't really understand why this step is needed per https://github.com/pi-hole/docker-pi-hole/issues/243 and if it needs to keep happening, or if version changes need to be requested in Debian before a full freeze occurs.

**Comment by lightswitch05** [comment]
Per [bullseye packages](https://packages.debian.org/testing/allpackages?format=txt.gz) libpam-modules-bin is supposed to already be `1.4.0-4`. Looks like the image is only 4 days old, but perhaps this will self-resolve in the next build

**Comment by lightswitch05** [comment]
I was right that the previous build failed due to a Debian package version mismatch and its now fixed upstream. Debian Bullseye is now in 'Hard Freeze' status. A full freeze is not yet determined: https://wiki.debian.org/DebianBullseye

**Comment by dschaper** [comment]
Any preferences for the tag version?

**Comment by lightswitch05** [comment]
I don't believe there is a need for a change of versions since this is a new image and will not affect existing images? If you still want to tag it, I have no prefereces

**Comment by dschaper** [comment]
If it's untagged then will it be used by any of the image consumers?

**Comment by dschaper** [comment]
Going by the https://github.com/pi-hole/docker-base-images#debian-base- docs and trying to understand where these fall in to that structure.

**Comment by dschaper** [comment]
And nevermind, I see clearly now. No change to any actual dockerfiles.

File: tag-and-push.sh [hunk 1]
```diff
@@ -15,8 +15,9 @@ function main() {
   # Push debian tags
   if [[ "${target:0:11}" = "debian-base" ]]; then
     flavor=${target:12:$$}
-    # TODO set buster as the primary version for `latest` and tag-based builds
-    primary_debian_version="stretch"
+    # Since this is a 'based images' repo, does it make sense to even provide
+    # a 'default' version? Perhaps all versions should be debian specific?
```
Comment (dschaper) [inline tag-and-push.sh:19]: Does this need to be addressed for follow-up?

Comment (lightswitch05) [reply tag-and-push.sh:19]: In docker-pi-hole the default is already set to specific Debian versions (buster currently). I know of anything using the 'latest' tag. https://github.com/pi-hole/docker-pi-hole/blob/20cd9c3abf95e00d2d1920ba432e2a0272c42942/build.yml#L20

**Comment by lightswitch05** [comment]
I believe this new image will get tagged as `bullseye` and `vX.X-bullseye`, which are all new tags, so shouldn't affect anything.

**Comment by lightswitch05** [comment]
This certainly isn't an urgent PR. My thoughts are that when new Debian releases come out, there is always in influx of people wanting it. I figured if there was already an 'alpha' image available whenever this happen, then it would be a great source of testing before it eventually becomes the new default image.

I don't feel like there was really enough community testing involved before when Buster became the default version and thought this might be a good way to improve on that.

**Comment by lightswitch05** [comment]
Bullseye is now in 'Hard Freeze'

> 2021-03-12: Hard Freeze - for key packages and packages without autopkgtests (scheduled) 
> 
> Full Freeze date not yet determined. 

Once this is merged, I'll create a followup PR for docker-pi-hole using the new bullseye image and we can determine if it 'just works' or if there will be followup work required to support the new Debian release.

**Comment by lightswitch05** [comment]
Any thoughts on getting this merged?

**Review by diginc — APPROVED** [review]

**Review by PromoFaux — APPROVED** [review]

**Pull request merged**